module/actor/witcherActor.js

```diff
@@ -25,7 +24,266 @@ export default class WitcherActor extends Actor {
+      isWounded = true
+      curRef = Math.floor((this.system.stats.ref.max + refTotalModifiers - armorEnc - encDiff) / 2 / refDivider)
+      curDex = Math.floor((this.system.stats.dex.max + dexTotalModifiers - armorEnc - encDiff) / 2 / dexDivider)
+      curInt = Math.floor((this.system.stats.int.max + intTotalModifiers) / 2 / intDivider)
+      curWill = Math.floor((this.system.stats.will.max + willTotalModifiers) / 2 / willDivider)
+    }
+
+    let hpTotalModifiers = this.getAllModifier("hp").totalModifiers;
+    let staTotalModifiers = this.getAllModifier("sta").totalModifiers;
+    let resTotalModifiers = this.getAllModifier("resolve").totalModifiers;
+    let focusTotalModifiers = this.getAllModifier("focus").totalModifiers;
+    let vigorModifiers = this.getAllModifier("vigor").totalModifiers;
+    let hpDivider = this.getAllModifier("hp").totalDivider;
+    let staDivider = this.getAllModifier("sta").totalDivider;
+    this.system.derivedStats.hp.modifiers.forEach(item => hpTotalModifiers += Number(item.value));
+    this.system.derivedStats.sta.modifiers.forEach(item => staTotalModifiers += Number(item.value));
+    this.system.derivedStats.resolve.modifiers.forEach(item => resTotalModifiers += Number(item.value));
+    this.system.derivedStats.focus.modifiers.forEach(item => focusTotalModifiers += Number(item.value));
+
+    let curHp = this.system.derivedStats.hp.max + hpTotalModifiers;
+    let curSta = this.system.derivedStats.sta.max + staTotalModifiers;
+    let curRes = this.system.derivedStats.resolve.max + resTotalModifiers;
+    let curFocus = this.system.derivedStats.focus.max + focusTotalModifiers;
+    let curVigor = this.system.derivedStats.vigor.unmodifiedMax + vigorModifiers;
+
+    let unmodifiedMaxHp = baseMax * 5
+
+    if (this.system.customStat != true) {
+      curHp = Math.floor((base * 5 + hpTotalModifiers) / hpDivider)
+      curSta = Math.floor((base * 5 + staTotalModifiers) / staDivider)
+      curRes = (Math.floor((curWill + curInt) / 2) * 5) + resTotalModifiers
+      curFocus = (Math.floor((curWill + curInt) / 2) * 3) + focusTotalModifiers
+    }
+
+    this.update({
+      'system.deathStateApplied': isDead,
+      'system.woundTresholdApplied': isWounded,
+      'system.stats.int.current': curInt,
+      'system.stats.ref.current': curRef,
+      'system.stats.dex.current': curDex,
+      'system.stats.body.current': curBody,
+      'system.stats.spd.current': curSpd,
+      'system.stats.emp.current': curEmp,
+      'system.stats.cra.current': curCra,
+      'system.stats.will.current': curWill,
+      'system.stats.luck.current': curLuck,
+
+      'system.derivedStats.hp.max': curHp,
+      'system.derivedStats.hp.unmodifiedMax': unmodifiedMaxHp,
+      'system.derivedStats.sta.max': curSta,
+      'system.derivedStats.resolve.max': curRes,
+      'system.derivedStats.focus.max': curFocus,
+      'system.derivedStats.vigor.max': curVigor,
```

Comment:
Once you have all values in maps, it's easy to convert them into these objects and put them here:

`...Array.from(statValuesMap.entries()).map(([stat, value]) => ({ [`system.stats.${stat}.current`]: value }));`

It is all just an idea how to make it easier. You might come up with a better idea or formats.

Reply:
https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/issues/61